consensus/src/quorum_store/proof_coordinator.rs

```diff
@@ -143,35 +185,36 @@ impl ProofCoordinator {
             self.init_proof(&signed_digest)?;
         }
         let digest = signed_digest.digest();
-
-        match self.digest_to_proof.entry(signed_digest.digest()) {
-            Entry::Occupied(mut entry) => {
-                entry.get_mut().add_signature(signed_digest)?;
-                if entry.get_mut().ready(validator_verifier) {
-                    let (_, state) = entry.remove_entry();
-                    let proof = state.take(validator_verifier);
-                    // quorum store measurements
-                    let duration = chrono::Utc::now().naive_utc().timestamp_micros() as u64
-                        - self
-                            .digest_to_time
-                            .remove(&digest)
-                            .expect("Batch created without recording the time!");
-                    counters::BATCH_TO_POS_DURATION
-                        .observe_duration(Duration::from_micros(duration));
-                    return Ok(Some(proof));
-                }
-            },
-            Entry::Vacant(_) => (),
+        if let Some(value) = self.digest_to_proof.get_mut(&signed_digest.digest()) {
+            value.add_signature(signed_digest, validator_verifier)?;
+            if !value.completed && value.ready(validator_verifier) {
+                let proof = value.take(validator_verifier);
+                // quorum store measurements
+                let duration = chrono::Utc::now().naive_utc().timestamp_micros() as u64
+                    - self
+                        .digest_to_time
+                        .remove(&digest)
+                        .expect("Batch created without recording the time!");
+                counters::BATCH_TO_POS_DURATION.observe_duration(Duration::from_micros(duration));
+                return Ok(Some(proof));
+            }
         }
         Ok(None)
     }
 
     async fn expire(&mut self) {
         let mut batch_ids = vec![];
         for signed_digest_info in self.timeouts.expire() {
-            counters::TIMEOUT_BATCHES_COUNT.inc();
```

Comment:
missing the counter?

Reply:
I'll add it back, but it is duplicate with counters::BATCH_SUCCESSFUL_CREATION